### PR TITLE
WIP: tests: completes test views

### LINKED ISSUE #804: Complete test views
The file https://github.com/inspirehep/inspire-next/blob/master/tests/test_views.py currently contains tests for the `/detailed` pages. 

Tests should be added for other main views such as:

* [ ] Landing pages (/literature, /jobs, ...)
* [ ] /admin/holdingpen/list (shall we create some objects and make sure they appear in the list ?)
* [ ] /author/new
* [ ] /author/update?recid=997876 ( so that imports data from that recid, otherwise it redirects to /new )
* [ ] /doi/search
* [ ] /arxiv/search
* [ ] /search with some different searches, sorting options, ...
* [ ] /submit/literature/create

Any others?

### PR DESCRIPTION
**Pull request by ioannistsanaktsidis**
* Closes #804.

Signed-off-by: Ioannis Tsanaktsidis <ioannis.tsanaktsidis@cern.ch>

### CODE REVIEW

File: tests/test_views.py [hunk 1]
```diff
@@ -36,3 +36,117 @@ def test_detailed_records(self):
...
+
+    def test_literature_landing_page(self):
+        """Test visiting each landing page returns 200"""
+        failing_landing_page = []
+        try:
+            if self.client.get("/literature").status_code != 200:
+                failing_landing_page.append('/literature')
+        except Exception:
+            failing_landing_page.append('/literature')
+        self.failIf(failing_landing_page)
```
Comment (jalavik) [inline tests/test_views.py]: Just a curiosity here, if an exception happens, will it be printed? (since it is caught before `failIf`)

Comment (ioannistsanaktsidis) [inline tests/test_views.py]: It will print sth like this:
['/literature'] as the above test prints a list with all the failing Ids. Makes sense?

**Comment by ioannistsanaktsidis** [comment]
@jmartinm this one fails due to the tests I added :P It is because of the flask toolbar the status code is 302 and then it redirects. Do we fix it here or in another PR?

**Comment by jmartinm** [comment]
I think Travis runs the tests with DEBUG=False so no debug toolbar should be present. Maybe it is something else? For example, if the page requires login you will be redirected to the login page.

**Comment by ioannistsanaktsidis** [comment]
True some of them require Login.

**Comment by jacquerie** [comment]
> True some of them require Login.

Maybe then adding `TESTING = True` will work for you, since it bypasses the `@login_required` decorator.

**Pull request closed without merging**